### PR TITLE
Refactor/best practice2(image component)

### PR DESCRIPTION
**Pull request by Ribhav-Jain22**
# Pull Request Template

## Description

refactored ImageHelper.ts + ImageConverter.vue + ImageModel.js for better reusability.

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Improvement (non-breaking change which improves code readability and/or UI Improvement)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [x] Yes
- [ ] No

## Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have fixed SonarCloud errors
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

File: vue-project/src/helpers/ImageHelper.ts [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+export function resizeImageByResolution(imageElement: HTMLImageElement, targetImageResolution: { width: number; height: number }, qualityFactor: number = 1.0): string {
+  const canvasElement = document.createElement('canvas');
+  canvasElement.width = targetImageResolution.width;
+  canvasElement.height = targetImageResolution.height;
+  const canvasElementContext = canvasElement.getContext('2d');
+  canvasElementContext.drawImage(imageElement, 0, 0, targetImageResolution.width, targetImageResolution.height);
+  return canvasElement.toDataURL('image/jpeg', qualityFactor);
+}
+
+export function resizedImageByFileSize(imageElement: HTMLImageElement, targetSizeInBytes: number): string {
```
Comment (ad-mittal) [inline vue-project/src/helpers/ImageHelper.ts]: resizeImageByFileSize
not 
resize**d**ImageByFileSize

Comment (ad-mittal) [reply vue-project/src/helpers/ImageHelper.ts]: Why does it return a string here?

File: vue-project/src/helpers/ImageHelper.ts [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+//FIXME rename imageElement to imageElement
+//FIXME rename targetWidth to targetWidthInPixels
+//FIXME rename targetHeight to targetHeightInPixels
+//FIXME combine targetHeight and targetWidth into one object targetSizeInPixels
+//FIXME rename canvasElement to canvasElementElement
+
+import { ImageResolution } from '@/models/image/ImageResolution';
+
+
+export function resizeImageByResolution(imageElement: HTMLImageElement, targetImageResolution: { width: number; height: number }, qualityFactor: number = 1.0): string {
```
Comment (ad-mittal) [inline vue-project/src/helpers/ImageHelper.ts:4]: why does this function return a string?

File: vue-project/src/helpers/ImageHelper.ts [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+}
+
+export function resizedImageByFileSize(imageElement: HTMLImageElement, targetSizeInBytes: number): string {
+  const minimumImageResolution = { width: 50, height: 50 }; // Minimum allowable dimensions to avoid too much reduction
+  let qualityFactor = 1.0;
+  const targetImageResolution = new ImageResolution(imageElement.width, imageElement.height);
+
+  // eslint-disable-next-line no-constant-condition
+  while (true) {
+    //2300x3400
```
Comment (ad-mittal) [inline vue-project/src/helpers/ImageHelper.ts]: remove this comment

File: vue-project/src/models/image/ImageModel.ts [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+  targetWidth: number
+  targetHeight: number
+  currentImageSrc: string;
+  currentFileSize: number;
+  currentFileName: string;
+  currentResolution: ImageResolution;
+
+  constructor(file) {
+
+    this.currentResolution = new ImageResolution(1, 1);
```
Comment (ad-mittal) [inline vue-project/src/models/image/ImageModel.ts]: This is wrong.
why are you setting defaults at all?
You can set all of these when the image is loaded

Comment (ad-mittal) [reply vue-project/src/models/image/ImageModel.ts]: This is wrong.
why are you setting defaults at all?
You can set all of these when the image is loaded

**Review by ad-mittal — CHANGES_REQUESTED** [review]

**Comment by Ribhav-Jain22** [comment]
I am working on it currently. Was fixing the ui issue.

**Review by ad-mittal — APPROVED** [review]

**Pull request merged**